lua/lspsaga/callhierarchy.lua

```diff
@@ -122,8 +122,14 @@ function ch:call_hierarchy(item, parent)
         icons[#icons + 1] = kind[target.kind]
         local expand_collapse = '  ' .. ui.expand
         local icon = kind[target.kind][2]
-        local className = target.uri:match('.+/(.+)[.]class[?]')
+
+        local projectClassPattern = '.+/([^/]+)[.]java'
```

Comment:
if this only for  java check buffer filetype as first ?

Reply:
Done - I have extracted the new functionality to be executed only if the filetype of the current buffer is java.
Left the classNamePart variable once we start implementing this for other languages as well.

fwiw, when I try to execute `Lspsaga outgoing_calls` in python or lua file (with LSP working), I get the following:
>method textDocument/prepareCallHierarchy is not supported by any of the servers registered for the current buffer

I will investigate further when I have more time. I have to fix this for my setup, so I can contribute for other languages as well (and handle them better whenever I add some functionality for java).

Thank you!